docs/guides/qiskit-code-assistant-local.mdx

```diff
@@ -0,0 +1,204 @@
+
+1. Try Ollama in your terminal and start running models e.g.,
+
+    ```
+    ollama run granite3-dense:8b
+    ```
+
+#### Set up Ollama with the Qiskit Code Assistant GGUF model
+
+1. Create a `Modelfile` entering the below content and be sure to update `<PATH-TO-GGUF-FILE>` to the actual path of your downloaded model
+
+    ```
+    FROM <PATH-TO-GGUF-FILE>
+    TEMPLATE """{{ if .System }}System:
+    {{ .System }}
+
+    {{ end }}
+    {{ if .Prompt }}Question:
+    {{ .Prompt }}
+
+    {{ end }}
+    Answer:
+    ```python
+    {{ .Response }}
+    """
+
+    SYSTEM """"""
+
+    PARAMETER stop "<fim_prefix>"
+    PARAMETER stop "<fim_middle>"
+    PARAMETER stop "<fim_suffix>"
+    PARAMETER stop "<fim_pad>"
+    PARAMETER stop "<|endoftext|>"
+
+    PARAMETER mirostat 0
+    PARAMETER mirostat_eta 0.1
+    PARAMETER mirostat_tau 5.0
+    PARAMETER num_ctx 10000
+    PARAMETER repeat_penalty 1.0
+    PARAMETER temperature 0.8
+    PARAMETER seed 0
+    PARAMETER tfs_z 1.0
+    PARAMETER num_predict 1024
+    PARAMETER top_k 50
+    PARAMETER top_p 0.95
+    PARAMETER min_p 0.05
+    ```
+
+1. Run the following command to create a custom model instance based on the `Modelfile`
+
+    ```
+    ollama create granite-8b-qiskit -f ./path-to-model-file
+    ```
+
+    <Admonition type="note">This process may take some time, Ollama will read the model file, initialize the model instance and configure it according to the specifications provided.</Admonition>
+
+
+#### Run the Qiskit Code Assistant model in Ollama
+
+After the `granite-8b-qiskit` GGUF model has been set up in Ollama, run the following command to launch the model and interact with it in the terminal (in chat mode)
+
+```
+ollama run granite-8b-qiskit
+```
+
+Some useful commands:
+
+- `ollama list` - List models on your computer
+- `ollama rm granite-8b-qiskit` - Remove/delete the model
+- `ollama show granite-8b-qiskit` - Show model information
+- `ollama stop granite-8b-qiskit` - Stop a model which is currently running
+- `ollama ps` - List which models are currently loaded
+
+### Using the `llama-cpp-python` package
+
+An alternative to the Ollama application is the `llama-cpp-python` package. It is a Python binding for `llama.cpp`. It gives you more control and flexibility to run the GGUF model locally. It’s ideal for users who wish to integrate the local model in their workflows and Python applications.
+
+1. Install `llama-cpp-python`: https://pypi.org/project/llama-cpp-python/
+1. Interact with the model from within your application using `llama_cpp` e.g.,
+
+```python
+from llama_cpp import Llama
+
+model_path = <PATH-TO-GGUF-FILE>
+
+model = Llama(
+        model_path, 
+        seed=17, 
+        n_ctx=10000,
+        n_gpu_layers=37, # to offload in gpu, but put 0 if all in cpu
+    )
+
+input = 'Generate a quantum circuit with 2 qubits'
+raw_pred = model(input)["choices"][0]["text"]
+```
+
+You can also add generate parameters to the model to customize the inference:
```

Comment:
Is "add generate parameters" an ok phrase or is it a typo?

Reply:
i updated to
```
You can also add text generation parameters to the model to customize the inference:
```
let me know if that is better?